src/renderer/uia/UiaRenderer.cpp

```diff
@@ -252,35 +288,34 @@ CATCH_RETURN();
         }
         CATCH_LOG();
     }
+    if (!_queuedOutput.empty())
+    {
+        try
+        {
+            // The speech API is limited to 1000 characters at a time.
+            // Break up the output into 1000 character chunks to ensure
+            // the output isn't cut off.
+            static const size_t sapiLimit{ 1000 };
+            while (_queuedOutput.size() > sapiLimit)
+            {
+                const auto croppedText{ _queuedOutput.substr(0, sapiLimit) };
+                _dispatcher->NotifyNewOutput(croppedText);
+                _queuedOutput = _queuedOutput.substr(sapiLimit);
+            }
+            _dispatcher->NotifyNewOutput(_queuedOutput);
+        }
+        CATCH_LOG();
+    }

```

Comment:
Use this to avoid copying the entire output string twice in each iteration:
```cpp
try
{
    static constexpr size_t sapiLimit = 1000;

    std::wstring_view output{ _queuedOutput }; // <-- std::wstring::substr returns a string and not a string_view

    for (size_t offset = 0;; offset += sapiLimit) // <-- only a single substr() needed if we use offset instead
    {
        const auto croppedText = output.substr(offset, sapiLimit); // <-- not copying
        if (croppedText.empty())
        {
            break;
        }

        _dispatcher->NotifyNewOutput(croppedText);
    }
}
CATCH_LOG()
```

(You can drop the if condition at the start.)